lessons/r/dplyrmagrittr/lesson.ipynb

```diff
@@ -0,0 +1,650 @@
+    "    arrange(Sepal.Length)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    select(Sepal.Length) %>%\n",
+    "    filter(Sepal.Length > 7.0) %>%\n",
+    "    arrange(desc(Sepal.Length))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "head(iris)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# arranging by x *then* y\n",
+    "\n",
+    "iris %>%\n",
+    "    select(Petal.Length:Species) %>%\n",
+    "    arrange(Petal.Length, Petal.Width)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `mutate` - create new columns"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    mutate(Sepal.Area = Sepal.Width * Sepal.Length,\n",
+    "          Petal.Area = Petal.Width * Petal.Length) %>%\n",
+    "    head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    filter(Species == 'setosa') %>%\n",
+    "    mutate(Length.Diff = Sepal.Length - Petal.Length) %>%\n",
+    "    select(Length.Diff) %>%\n",
+    "    head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `group_by` and `mutate` - create new variables in a grouped fashion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    group_by(Species) %>%\n",
+    "    mutate(Sepal.Length.Deviation = mean(Sepal.Length) - Sepal.Length) %>%\n",
+    "    head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    mutate(Sepal.Area = Sepal.Width * Sepal.Length,\n",
+    "           Petal.Area = Petal.Width * Petal.Length) %>%\n",
+    "    group_by(Species) %>%\n",
+    "    mutate(Grouped.Area.Ratio = Sepal.Area / mean(Sepal.Area)) %>%\n",
+    "    head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `summarise` - summarise your data\n",
+    "\n",
+    "# `group_by` and `summarise` - summarise your data in a grouped fashion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    group_by(Species) %>%\n",
+    "    summarise(n = n())"
```

Comment:
I do use `tally`. They are used for different purposes, though I believe in this case they output the same.